gems/sorbet-runtime/lib/types/types/opus_enum.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+# typed: true
+
+module T::Types
+  # Validates that an object is equal to another Opus::Enum singleton value.
+  class OpusEnum < Base
+    attr_reader :val
+
+    def initialize(val)
+      @val = val
+    end
+
+    # @override Base
+    def name
+      @val.inspect
+    end
+
+    # @override Base
+    def valid?(obj)
+      @val == obj
+    end
+
+    # @override Base
+    private def subtype_of_single?(other)
+      case other
+      when OpusEnum
```

Comment:
Does this mean that you can compare values of different enums, and potentially get a result of `true`?

Reply:
I don't think so. This means that the reified type wrapper we're comparing against must be a `T::Types::OpusEnum` wrapper. If it isn't that, then it's always false. If it is that, then we know that the wrapper has a `val` reader that we can use to get the underlying `::Opus::Enum` being wrapped, and subtype_of_single only is valid if our val is directly equal to their val.